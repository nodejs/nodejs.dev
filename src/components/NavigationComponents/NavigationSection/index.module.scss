@import '../../../styles/mixins.scss';

.navigation-section,
%navigation-section {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;

  @media (max-width: 900px) {
    height: auto;
    overflow: visible;
    top: 24px;
  }

  &:first-of-type {
    margin-top: var(--space-40);

    @media (max-width: 900px) {
      margin-top: 0;
    }
  }
}

.navigation-section-title,
%navigation-section-title {
  @include navigationItemBase;

  border: none;
  color: var(--color-text-primary);
  display: flex;
  flex-direction: row;
  font-weight: var(--font-weight-semibold);
  justify-content: space-between;
  margin: var(--space-24) 0 var(--space-16) var(--space-24);
  width: 100%;

  > span {
    align-items: center;
    display: flex;
  }
}
