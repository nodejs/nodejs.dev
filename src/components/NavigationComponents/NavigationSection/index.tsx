import React from 'react';
import styles from './index.module.scss';

interface Props {
  isOpen: boolean;
  label: string;
  title: React.ReactNode;
  content: React.ReactNode;
}

const getDisplay = (isOpen: boolean) => ({
  display: isOpen ? 'block' : 'none',
});

const NavigationSection = ({
  label,
  title,
  isOpen,
  content,
}: Props): JSX.Element => (
  <div className={styles.navigationSection}>
    {title}
    <div role="region" aria-label={label} style={getDisplay(isOpen)}>
      {content}
    </div>
  </div>
);

export default NavigationSection;
