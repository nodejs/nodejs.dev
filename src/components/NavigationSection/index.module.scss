@import '../NavigationItem/index.module.scss';

.navigation-section,
%navigation-section {
  $section: &;
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;

  @media (max-width: 800px) {
    height: auto;
    overflow: visible;
    top: 24px;
  }

  &:first-of-type {
    margin-top: var(--space-40);
  }

  &__pure-title,
  &%__pure-title {
    color: var(--color-text-primary);
    display: flex;
    flex-direction: row;
    font-weight: var(--font-weight-semibold);
    justify-content: space-between;
    margin: var(--space-24) 0 var(--space-16) var(--space-24);

    span {
      align-items: center;
      display: flex;
    }

    i {
      padding: 0;
    }

    &::before,
    &:hover::before {
      box-shadow: inset 0 0 0 3px var(--color-text-primary);
    }

    &::after {
      background: transparent;
    }
  }

  &__title,
  &%__title {
    @extend %navigation-item;
    @extend %navigation-section__pure-title;
  }

  &--api {
    &__title {
      margin: 0;

      span svg {
        font-size: 20px;
        padding-left: 0;
      }

      &::before,
      &:hover::before {
        box-shadow: none;
        content: '';
        height: 0;
        padding: 0;
        width: 0;
      }
    }
  }
}
