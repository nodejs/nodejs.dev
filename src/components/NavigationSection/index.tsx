import React from 'react';
import { NavigationSectionItem } from '../../types';
import NavigationItem from '../NavigationItem';

interface Props {
  key: string;
  title: string;
  section: NavigationSectionItem[];
  currentSlug: string;
  onItemClick: () => void;
  readSections: Set<NavigationSectionItem['slug']>;
}

const NavigationSection = ({
  title,
  section,
  currentSlug,
  onItemClick,
  readSections,
}: Props): JSX.Element => {
  return (
    <ul className="side-nav__list">
      <h2 className="t-body2 side-nav__title">
        <i className="material-icons">offline_bolt</i>
        {title}
      </h2>
      {section.map(
        (item: NavigationSectionItem): JSX.Element => {
          const isRead: boolean = readSections.has(item.slug);

          return (
            <NavigationItem
              key={item.slug}
              title={item.title}
              slug={item.slug}
              isRead={isRead}
              isActive={item.slug === currentSlug}
              onClick={onItemClick}
            />
          );
        }
      )}
    </ul>
  );
};

export default NavigationSection;
