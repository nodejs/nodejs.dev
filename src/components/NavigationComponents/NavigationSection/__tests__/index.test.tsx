import React from 'react';
import { render } from '@testing-library/react';
import NavigationSection from '../index';

describe('NavigationSection component', (): void => {
  it('renders correctly', (): void => {
    const label = 'API Navigation Section';
    const isOpen = false;
    const mockTitle = <h1>Mock Title</h1>;
    const mockContent = <p>Mock content</p>;
    const { container } = render(
      <NavigationSection
        isOpen={isOpen}
        label={label}
        title={mockTitle}
        content={mockContent}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
