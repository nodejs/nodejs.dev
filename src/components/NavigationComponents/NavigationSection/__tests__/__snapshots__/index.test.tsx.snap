// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationSection component renders correctly 1`] = `
<div>
  <div
    class="navigationSection"
  >
    <h1>
      Mock Title
    </h1>
    <div
      aria-label="API Navigation Section"
      role="region"
      style="display: none;"
    >
      <p>
        Mock content
      </p>
    </div>
  </div>
</div>
`;
